contracts/FungibleTokenSwitchboard.cdc

```diff
@@ -0,0 +1,143 @@
+            // Borrow a reference to the vault pointed by the capability we want
+            // to store inside the switchboard
+            let vaultRef = capability.borrow() 
+                ?? panic ("Cannot borrow reference to vault from capability")
+            // Use the vault reference type as key for storing the capability
+            self.receiverCapabilities[vaultRef.getType()] = capability
+            // Emit the event that indicates that a new capability has been added
+            emit VaultCapabilityAdded(type: vaultRef.getType())
+        }
+
+        /// removeVault removes a fungible token receiver capability 
+        ///                       from the switchboard resource
+        /// 
+        /// Parameters: capability: The capability to a fungible token vault 
+        ///                         to be removed from the switchboard
+        ///
+        pub fun removeVault(capability: Capability<&{FungibleToken.Receiver}>) {
+            // Borrow a reference to the vault pointed by the capability we want
+            // store inside the switchboard            
+            let vaultRef = capability.borrow() 
+                ?? panic ("Cannot borrow reference to vault from capability")
+            // Use the vault reference to find the capability to remove
+            self.receiverCapabilities.remove(key: vaultRef.getType())
+            // Emit the event that indicates that a new capability has been removed
+            emit VaultCapabilityRemoved(type: vaultRef.getType())            
+        }
+        
+        /// deposit Takes a fungible token vault and routes it to the proper
+        ///         fungible token receiver capability for depositing it
+        /// 
+        /// Parameters: from: The deposited fungible token vault resource
+        ///        
+        pub fun deposit(from: @FungibleToken.Vault) {
+            let depositedVaultCapability = self.receiverCapabilities[from.getType()] ?? 
+                panic ("The deposited vault is not available on this switchboard")
+            let vaultRef = depositedVaultCapability.borrow() ?? 
+                panic ("Can not borrow a reference to the the vault")
+            vaultRef.deposit(from: <- from)
+        }
+
+        /// getVaultCabilities function for get to know which tokens a certain
+        /// switchboard resource is prepared to receive
+        ///
+        /// Returns: The dictionary of stored {FungibleToken.Receiver} 
+        /// capabilities
+        ///
+        pub fun getVaultCapabilities(): {Type: Capability<&{FungibleToken.Receiver}>} {
+            return self.receiverCapabilities
+        }
+
+        init() {
+            // Initialize the capabilities dictionary
+            self.receiverCapabilities = {}
+            // Emit the event that indicates that a new switchboard resource
+            // has been created
+            emit SwitchboardInitialized(switchboardResourceID: self.uuid)
```

Comment:
is this event has any use? 

I think technically it is good practice to avoid such events if only information we can share is `uuid`. But I may be missing something.

Reply:
yeah, probably can remove this

Reply:
Yeah, I totally agree that's not really providing any value